src/pages/UserPage/UserPage.jsx

```diff
@@ -1,96 +1,204 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import Header from '../../components/HeaderComponent/HeaderComponent';
 import Footer from '../../components/Footer/Footer';
 import './UserPage.css';
-import userDB from '../../assets/admins.json';
 import { RiEdit2Fill } from 'react-icons/ri'
-
-
+import { useParams } from 'react-router';
+import languageSelector from '../../assets/languages/languageSelector';
+import { useSelector, useDispatch } from 'react-redux';
+import loadingGif from '../../assets/fotos/loading/loading-gif.gif'
+import axios from 'axios';
+import {
+  fetchUserRequest,
+  fetchUserSuccess,
+  fetchUserFailure,
+  updateUserRequest,
+  updateUserSuccess,
+  updateUserFailure
+} from '../../store/actions/user.action';
 
 const UserPage = () => {
-  const [user, setUser] = useState(userDB[0]);
+  const { id } = useParams();
+  const language = useSelector(state => state.languageReducer);
+  const loading = useSelector((state) => state.userReducer.loading);
+  const user = useSelector(state => state.userReducer.user);
   const [isEditable, setIsEditable] = useState(false);
   const [errors, setErrors] = useState({});
+  const dispatch = useDispatch();
+  
   const [formData, setFormData] = useState({
-    name: user.name,
-    email: user.email,
-    password: user.password,
-    phoneNumber: user.phoneNumber,
-    address: user.address,
-    city: user.city
+    name: '',
+    last_name: '',
+    email: '',
+    password: '',
+    phone_numbers: [],
+    address: [],
+    city: ''
   });
 
+
+  // does a get request on mount 
+  useEffect(() => {
+    const fetchUser = async () => {
+      dispatch(fetchUserRequest());
+      try {
+        const response = await axios.get(`http://localhost:8080/api/users/${id}`);
+        dispatch(fetchUserSuccess(response.data.data));
+        setFormData({
+          name: response.data.data.name,
+          last_name: response.data.data.last_name,
+          email: response.data.data.email,
+          password: response.data.data.password,
+          phone_numbers: response.data.data.phone_numbers,
+          address: response.data.data.addresses,
+          city: response.data.data.city,
```

Comment:
consider a possible enhancement: destructure the response the double date object or rename one of the data objects and finally, destructure it to get the props for the object, could be better yet and easier for maintenance.

Reply:
Resolved, but changing addres causes the app to crash

Reply:
Got it, but you covered well, by the way
```js
     const { 
          name, 
          last_name, 
          email, 
          password, 
          phone_numbers, 
          addresses: address, 
          city 
        } = response.data.data;
```
This would work, let's move it to resolved and test this approach later, good one.